client/db/bolt/db.go

```diff
@@ -498,7 +503,7 @@ func decodeOrderBucket(oid []byte, oBkt *bbolt.Bucket) (*dexdb.MetaOrder, error)
 	}
 
 	var linkedID order.OrderID
-	copy(linkedID[:], getCopy(oBkt, linkedKey))
```

Comment:
Note that now that we are storing `nil`, this `Get` will return `[]byte{}` instead of `nil`.  That's OK since there are no bytes to copy and it stays the zero `OrderID`.  Just wanted to point that distinction out.  I think that's what you envisioned for the decode now, right?